### PR TITLE
Doc updates around DIALECT

### PR DESCRIPTION
**Pull request by slorello89**
The VSS docs and the FT.SEARCH command docs don't explain the need to set DIALECT to 2 in order to use the PARAMS.

### CODE REVIEW

File: docs/commands/ft.search.md [hunk 1]
```diff
@@ -76,7 +76,9 @@ Searches the index with a textual query, returning either documents or just ids.
...
+* **PARAMS {nargs} {name} {value}**. Define one or more value parameters. Each parameter has a name and a value. Parameters can be referenced in the **query** by a `$`, followed by the parameter name, e.g., `$user`, and each such reference in the search query to a parameter name is substituted by the corresponding parameter value. For example, with parameter definition `PARAMS 4 lon 29.69465 lat 34.95126`, the expression `@loc:[$lon $lat 10 km]` would be evaluated to `@loc:[29.69465 34.95126 10 km]`. Parameters cannot be referenced in the query string where concrete values are not allowed, such as in field names, e.g., `@loc`. To use `PARAMS`, `DIALECT` must be set to 2.
+
+- **DIALECT**: The query syntax to use for the query. Options are 1 and 2.
```
Comment (oshadmi) [inline docs/commands/ft.search.md:81]: These additions seem to have been lost during the move of the documentation to redis.io

https://github.com/RediSearch/RediSearch/pull/2671/files#diff-13483592b824255998c18c26ba13d15aaa8cf9aac19d0135e07000204a0cbaabR462

Also relevant in `FT.AGGREGATE`, `FT.EXPLAIN`, `FT.EXPLAINCLI`, `FT.SPELLCHECK`

**Review by oshadmi — DISMISSED** [review]
👍🏼 Looks good!

File: docs/docs/reference/Vectors.md [hunk 2]
```diff
@@ -149,7 +149,7 @@ FT.CREATE ... SCHEMA ... {field_name} VECTOR {algorithm} {count} [{attribute_nam
 
 ## Querying vector fields
 
-We allow using vector similarity queries in the `FT.SEARCH` "query" parameter. The syntax for vector similarity queries is `*=>[{vector similarity query}]` for running the query on an entire vector field, or `{primary filter query}=>[{vector similarity query}]` for running similarity query on the result of the primary filter query.
+We allow using vector similarity queries in the `FT.SEARCH` "query" parameter. The syntax for vector similarity queries is `*=>[{vector similarity query}]` for running the query on an entire vector field, or `{primary filter query}=>[{vector similarity query}]` for running similarity query on the result of the primary filter query. To use a vector similarity query, you must set the `DIALECT` option to 2.
```
Comment (oshadmi) [inline docs/docs/reference/Vectors.md:152]:
```suggestion
We allow using vector similarity queries in the `FT.SEARCH` "query" parameter. The syntax for vector similarity queries is `*=>[{vector similarity query}]` for running the query on an entire vector field, or `{primary filter query}=>[{vector similarity query}]` for running similarity query on the result of the primary filter query. To use a vector similarity query, you must specify the option `DIALECT 2` in the command itself, or by setting the `DEFAULT_DIALECT` option to `2`, either with the command`FT.CONFIG SET` or when loading the `redisearch` module and passing it the argument `DEFAULT_DIALECT 2`.
```

**Review by oshadmi — COMMENTED** [review]
On second thought, some addition might be required

**Review by DvirDukhan — APPROVED** [review]

**Review by rafie — APPROVED** [review]

**Pull request merged**